stdlib/influxdata/influxdb/from.go

```diff
@@ -69,3 +147,300 @@ func (s *FromProcedureSpec) Copy() plan.ProcedureSpec {
+		ns.Transformations = make([]plan.ProcedureSpec, 0, len(s.Transformations)+1)
+		copy(ns.Transformations, s.Transformations)
+	}
+	return ns
+}
+
+type source struct {
+	id   execute.DatasetID
+	spec *FromRemoteProcedureSpec
+	deps flux.Dependencies
+	mem  *memory.Allocator
+	ts   execute.TransformationSet
+}
+
+func createFromSource(ps plan.ProcedureSpec, id execute.DatasetID, a execute.Administration) (execute.Source, error) {
+	spec := ps.(*FromRemoteProcedureSpec)
+	if spec.Range == nil {
+		return nil, errors.Newf(codes.Invalid, "bounds must be set")
+	}
+
+	// These parameters are only required for the remote influxdb
+	// source. If running flux within influxdb, these aren't
+	// required.
+	if spec.Org == nil {
+		return nil, errors.Newf(codes.Invalid, "org must be set")
+	}
+
+	deps := flux.GetDependencies(a.Context())
+	s := &source{
+		id:   id,
+		spec: spec,
+		deps: deps,
+		mem:  a.Allocator(),
+	}
+
+	if err := s.validateHost(*spec.Host); err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
+func (s *source) AddTransformation(t execute.Transformation) {
+	s.ts = append(s.ts, t)
+}
+
+func (s *source) Run(ctx context.Context) {
+	err := s.run(ctx)
+	s.ts.Finish(s.id, err)
+}
+
+func (s *source) run(ctx context.Context) error {
+	req, err := s.newRequest(ctx)
+	if err != nil {
+		return err
+	}
+
+	client, err := s.deps.HTTPClient()
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	} else if resp.StatusCode != 200 {
+		data, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return errors.Newf(codes.Invalid, "error when reading response body: %s", err)
+		}
+		return s.parseError(data)
+	}
+	return s.processResults(resp.Body)
+}
+
+func (s *source) validateHost(host string) error {
+	validator, err := s.deps.URLValidator()
+	if err != nil {
+		return err
+	}
+
+	u, err := url.Parse(host)
+	if err != nil {
+		return err
+	}
+	return validator.Validate(u)
+}
+
+func (s *source) newRequest(ctx context.Context) (*http.Request, error) {
+	u, err := url.Parse(*s.spec.Host)
+	if err != nil {
+		return nil, err
+	}
+	u.Path += "/api/v2/query"
+	u.RawQuery = func() string {
+		params := make(url.Values)
+		if s.spec.Org.ID != "" {
+			params.Set("orgID", s.spec.Org.ID)
+		} else {
+			params.Set("org", s.spec.Org.Name)
+		}
+		return params.Encode()
+	}()
+
+	// Validate that the produced url is allowed.
+	urlv, err := s.deps.URLValidator()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := urlv.Validate(u); err != nil {
+		return nil, err
+	}
+
+	body, err := s.newRequestBody()
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("POST", u.String(), bytes.NewReader(body))
+	if err != nil {
+		return nil, err
+	}
+	if s.spec.Token != nil {
+		req.Header.Set("Authorization", "Token "+*s.spec.Token)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	return req.WithContext(ctx), nil
+}
+
+func (s *source) newRequestBody() ([]byte, error) {
```

Comment:
Wondering if all  this logic should be sometime in the future embedded in our 2.x go client.

For now, it only allows to query with a `string` (no `ast`), and returns raw results or deserialized into a structure whose field names matches the column names.

If we add the `ast` kind of  request to that, `flux` could depend on that and decode the raw response.

Reply:
It should, but to my knowledge, the client is in the influxdb repository and we can't depend on it. That's why there's some duplication that I am hoping eventually can be removed. But the http code to make the queries isn't too difficult on its own and most of it is in the flux repository so it was easier to redo the small section that was in influxdb.

Reply:
I think the client is in a separate repo: https://github.com/influxdata/influxdb-client-go

Anyways, it was just a suggestion for the future, nothing that should block this PR